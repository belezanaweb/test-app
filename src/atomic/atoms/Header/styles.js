import styled from 'styled-components/native';
import {Platform} from 'react-native';

import colors from '../../constants/colors';

export const BackButton = styled.TouchableOpacity`
  top: ${Platform.OS === 'ios' ? 54 : 24}px 
  left: 20px;
  position: absolute;
`;

export const SnackBar = styled.View`
  background-color: ${colors.gold};
  padding-top: 15px;
  top: 0px;
  left: 0px;
  height: 40px;
  position: absolute;
  z-index: 9999;
  width: 100%;
  justify-content: center;
  align-items: center;
`;
