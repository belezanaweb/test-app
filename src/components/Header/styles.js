import EStyleSheet from 'react-native-extended-stylesheet';
export default EStyleSheet.create({
  container: {
    backgroundColor: '$lightGray',
    justifyContent: 'space-around',
    alignItems: 'center',
    padding: 5,
    paddingTop: 10,
    flexDirection: 'row'
  },
  safeArea: {
    backgroundColor: '$lightGray'
  },
  title: {
    color: '$dark',
    fontWeight: 'bold',
    fontSize: 16,
    textAlign: 'center'
  },

  containerLeft: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'flex-start',
    marginLeft: 5
  },
  containerCenter: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    flex: 1
  }
});
