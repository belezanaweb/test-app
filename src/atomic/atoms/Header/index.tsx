import React, { useState, useEffect } from 'react'
import { View, Platform, StatusBar, Switch } from 'react-native'

import { connect } from 'react-redux'
import * as appAction from '../../../redux/actions/appActions'

import { useNetInfo } from '@react-native-community/netinfo'

import colors from '../../constants/colors'

import { TextRegular } from '../../atoms/Titles'
import Icons from '../../atoms/Icons'

import { BackButton, SnackBar } from './styles'

function Header({ navigation, title, backButton, _setDarkMode, darkMode }) {
  //handling with net
  const netInfo = useNetInfo()

  const toggleSwitch = () => {
    //setDarkMode((previousState) => !previousState);
    _setDarkMode(!darkMode)
  }

  const showDarkModeButton = false

  return (
    <>
      <StatusBar barStyle={darkMode ? 'light-content' : 'dark-content'} />

      {!netInfo.isConnected && (
        <SnackBar>
          <TextRegular size={12}>
            <Icons name="wifi" size={12} />
            &nbsp;No internet connection available
          </TextRegular>
        </SnackBar>
      )}

      <View
        style={{
          flexDirection: 'row',
          backgroundColor: darkMode ? colors.graffit : colors.lightGray,
          padding: 20,
          paddingTop: Platform.OS === 'ios' ? 50 : 20,
          justifyContent: 'center',
          position: 'relative'
        }}
      >
        {backButton && (
          <BackButton
            onPress={() => {
              navigation.goBack()
            }}
          >
            <Icons name="chevron-left" size={18} color={darkMode ? colors.orange : colors.black} />
          </BackButton>
        )}

        <TextRegular weight={'bold'} size={14} align={'center'} color={darkMode ? colors.orange : colors.black}>
          {title}
        </TextRegular>

        <View
          style={{
            flexDirection: 'row',
            position: 'absolute',
            bottom: 17,
            right: 20
          }}
        >
          {showDarkModeButton && (
            <Switch
              trackColor={{ false: colors.gray, true: colors.gray }}
              thumbColor={darkMode ? colors.orange : colors.white}
              ios_backgroundColor={colors.gray}
              onValueChange={toggleSwitch}
              value={darkMode}
            />
          )}
        </View>
      </View>
    </>
  )
}

const mapStateToProps = state => ({
  darkMode: state.appReducer.darkMode
})

const mapDispatchToProps = dispatch => {
  return {
    _setDarkMode: bool => {
      dispatch(appAction.AppSetDarkMode(bool))
    }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Header)
