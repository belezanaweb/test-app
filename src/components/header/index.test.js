import 'jest-styled-components';
// do not forget the /native at the end of the line.
import React from 'react';
import renderer from 'react-test-renderer';
import Header from './index';

describe('Component Header Snapshot', () => {
	test('Component Header Snapshot', () => {
		const tree = renderer.create(<Header close={false} name="Test" />).toJSON();

		expect(tree).toMatchSnapshot();
	});
});
