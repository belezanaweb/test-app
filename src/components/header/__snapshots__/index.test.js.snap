// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Header Snapshot Component Header Snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Roboto-Medium",
          "fontSize": 20,
          "lineHeight": 20,
          "textAlign": "center",
          "textTransform": "uppercase",
          "width": "100%",
        },
      ]
    }
  >
    Test
  </Text>
</View>
`;
